### ISSUE TITLE
NuGet pack .csproj with packages.config is ignoring dependency packages

### ISSUE THREAD

**Issue by Tasteful** [NuGet/Home#2862]
I using the beta version `NuGet Version: 3.5.0.1365` after to packing problem with released version and .net core RC2.

The nupkg that is produced from an csproj file that have a packages.config file with external dependency is missing all the external dependency.

```
C:\tfs\CompanyProduct.Dev>"C:\tfs\CompanyProduct.Dev\.nuget\NuGet.exe" pack "C:\tfs\CompanyProduct.Dev\Src\Company.Infrastructure.InversionOfControl\Company.Infrastructure.InversionOfControl.csproj" -NoDefaultExcludes -Properties Configuration=Release;LibFolder=lib\net461;OutputPath=bin\Release -NonInteractive -OutputDirectory "C:\tfs\CompanyProduct.Dev\Builds" -verbosity detailed
NuGet Version: 3.5.0.1365
Attempting to build package from 'Company.Infrastructure.InversionOfControl.csproj'.
MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
Packing files from 'C:\tfs\CompanyProduct.Dev\Src\Company.Infrastructure.InversionOfControl\bin\Release'.
Using 'Company.Infrastructure.InversionOfControl.nuspec' for metadata.
Add file 'C:\tfs\CompanyProduct.Dev\Src\Company.Infrastructure.InversionOfControl\bin\Release\Company.Infrastructure.InversionOfControl.dll' to package as 'lib\net461\Company.Infrastructure.InversionOfControl.dll'
Add file 'C:\tfs\CompanyProduct.Dev\Src\Company.Infrastructure.InversionOfControl\bin\Release\Company.Infrastructure.InversionOfControl.XML' to package as 'lib\net461\Company.Infrastructure.InversionOfControl.XML'
Found packages.config. Using packages listed as dependencies

Id: Company.Infrastructure.InversionOfControl
Version: 5.0.0-p160530082256
Authors: Company AB
Description: Castle Windsor Inversion Of Control implementation for Company.Owin
Dependencies:

Added file '[Content_Types].xml'.
Added file '_rels/.rels'.
Added file 'lib/net461/Company.Infrastructure.InversionOfControl.dll'.
Added file 'lib/net461/Company.Infrastructure.InversionOfControl.pdb'.
Added file 'lib/net461/Company.Infrastructure.InversionOfControl.XML'.
Added file 'Company.Infrastructure.InversionOfControl.nuspec'.
Added file 'package/services/metadata/core-properties/52ffeef0c9ad475b8a9ad9f6865efccc.psmdcp'.

Successfully created package 'C:\tfs\CompanyProduct.Dev\Builds\Company.Infrastructure.InversionOfControl.5.0.0-p160530082256.nupkg'.
```

Packages.config

```
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="Castle.Core" version="3.3.3" targetFramework="net461" />
  <package id="Castle.Windsor" version="3.3.0" targetFramework="net461" />
  <package id="Microsoft.Extensions.DependencyInjection.Abstractions" version="1.0.0-rc2-final" targetFramework="net461" />
  <package id="Newtonsoft.Json" version="8.0.3" targetFramework="net461" developmentDependency="true" />
  <package id="System.ComponentModel" version="4.0.1-rc2-24027" targetFramework="net461" />
  <package id="System.Diagnostics.Debug" version="4.0.11-rc2-24027" targetFramework="net461" />
  <package id="System.Globalization" version="4.0.11-rc2-24027" targetFramework="net461" />
  <package id="System.Linq" version="4.1.0-rc2-24027" targetFramework="net461" />
  <package id="System.Linq.Expressions" version="4.0.11-rc2-24027" targetFramework="net461" />
  <package id="System.Reflection" version="4.1.0-rc2-24027" targetFramework="net461" />
  <package id="System.Resources.ResourceManager" version="4.0.1-rc2-24027" targetFramework="net461" />
</packages>
```

Company.Infrastructure.InversionOfControl.nuspec

```
<?xml version="1.0"?>
<package >
    <metadata>
        <id>$id$</id>
        <version>$version$</version>
        <authors>$author$</authors>
        <owners>$author$</owners>
        <description>$description$</description>
        <copyright>$copyright$</copyright>
        <requireLicenseAcceptance>false</requireLicenseAcceptance>
    </metadata>
    <files>
        <file src="$OutputPath$\$id$.pdb" target="$LibFolder$" />
    </files>
</package>
```

Nuspec from the generated nupkg

```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2011/10/nuspec.xsd">
  <metadata>
    <id>Company.Infrastructure.InversionOfControl</id>
    <version>5.0.0-p160530082256</version>
    <authors>Company AB</authors>
    <owners>Company AB</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <description>Castle Windsor Inversion Of Control implementation for Company.Owin</description>
    <copyright>Copyright © 2016 Company AB</copyright>
    <dependencies />
  </metadata>
</package>
```

**Comment by Tasteful** [comment]
the problem seams to be around this check https://github.com/NuGet/NuGet.Client/blob/bd9b4551438897e122d66e490ba6466c59e03a6c/src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs#L986 that says that the dependency nupkg file should exists directly in a packages-subdirectory to project dir (ex Product\src\Company.Library\Packages where Company.Library is the project directory)

Checkin https://github.com/NuGet/NuGet.Client/commit/bd9b4551438897e122d66e490ba6466c59e03a6c 

ping @toddm

**Comment by Tasteful** [comment]
Whas trying with other release from myget and https://www.myget.org/feed/nugetbuild/package/nuget/NuGet.CommandLine/3.5.0-rc-1315 was working as expected with the dependecies.

**Comment by rrelyea** [comment]
Sounds to me like this is fixed in current builds of dev branch. But broken in 3.5.0-beta.
Should consider port to 3.5.0-beta2

**Comment by Tasteful** [comment]
@rrelyea The bug exists in the latest release on myget (https://www.myget.org/feed/nugetbuild/package/nuget/NuGet.CommandLine/3.5.0-rc-1365)

Does it exists any other feed with a more recent version that I can test and verify if the issue still is there?

**Comment by toddm** [comment]
That is the latest.  I think it's broken in the latest build.  I'll take a look at it.

**Comment by toddm** [comment]
If you pack with an earlier build like the latest 3.4.4 build, what dependencies end up in the nuspec file in the nupkg?
I'm trying to reproduce the problem with a test project but I get the same result with earlier versions of NuGet.exe so I'm not yet seeing the issue locally.

**Comment by TimBarham** [comment]
@toddm the logic to find the package seems wrong to me. For example, if I restore a project in VS then pack from the command line, it won't find the packages (since they will be in a `packages` directory off the solution directory, and we don't look there). And the nuspec file will be in a folder named for the package, which pack doesn't look for. So we can't just assume it is in a `packages` directory in the project directory. The `restore` command has example logic for locating the packages directory, and will fail with an error message if it can't find it. Perhaps `pack` should do the same, rather than failing silently when it can't find packages?

**Comment by Tasteful** [comment]
@toddm 

Pack result with nuget 3.4.3.855 and 3.4.4.870

```
Attempting to build package from 'Company.Infrastructure.InversionOfControl.csproj'.
MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
Packing files from 'C:\tfs\CompanyProduct.Dev\Src\Company.Infrastructure.InversionOfControl\bin\Release'.
Using 'Company.Infrastructure.InversionOfControl.nuspec' for metadata.
Add file 'C:\tfs\CompanyProduct.Dev\Src\Company.Infrastructure.InversionOfControl\bin\Release\Company.Infrastructure.InversionOfControl.dll' to package as 'lib\net461\Company.Infrastructure.InversionOfControl.dll'
Add file 'C:\tfs\CompanyProduct.Dev\Src\Company.Infrastructure.InversionOfControl\bin\Release\Company.Infrastructure.InversionOfControl.XML' to package as 'lib\net461\Company.Infrastructure.InversionOfControl.XML'
System.Xml.XmlException: The ',' character, hexadecimal value 0x2C, cannot be included in a name. Line 5, position 27.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.Throw(String res, String[] args)
   at System.Xml.XmlTextReaderImpl.Throw(Int32 pos, String res, String[] args)
   at System.Xml.XmlTextReaderImpl.ParseElement()
   at System.Xml.XmlTextReaderImpl.ParseElementContent()
   at System.Xml.XmlTextReaderImpl.Read()
   at System.Xml.XmlTextReader.Read()
   at MS.Internal.IO.Packaging.PartBasedPackageProperties.ParseCorePropertyPart(PackagePart part)
   at MS.Internal.IO.Packaging.PartBasedPackageProperties.ReadPropertyValuesFromPackage()
   at MS.Internal.IO.Packaging.PartBasedPackageProperties..ctor(Package package)
   at System.IO.Packaging.Package.get_PackageProperties()
   at NuGet.OptimizedZipPackage.<>c__DisplayClassa.<EnsurePackageFiles>b__9(PackagePart part)
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at NuGet.OptimizedZipPackage.EnsurePackageFiles()
   at NuGet.OptimizedZipPackage.GetFilesBase()
   at NuGet.LocalPackage.GetFiles()
   at NuGet.PackageExtensions.GetFiles(IPackage package, String directory)
   at NuGet.CommandLine.ProjectFactory.<>c.<AddFiles>b__92_2(IPackage a)
   at System.Linq.Enumerable.<SelectManyIterator>d__16`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at NuGet.CommandLine.ProjectFactory.AddFiles(PackageBuilder builder, String itemType, String targetFolder)
   at NuGet.CommandLine.ProjectFactory.<>c__DisplayClass56_0.<CreateBuilder>b__1(ProjectFactory p)
   at NuGet.CommandLine.ProjectFactory.ApplyAction(Action`1 action)
   at NuGet.CommandLine.ProjectFactory.CreateBuilder(String basePath)
   at NuGet.CommandLine.PackCommand.BuildFromProjectFile(String path)
   at NuGet.CommandLine.PackCommand.BuildPackage(String path)
   at NuGet.CommandLine.PackCommand.ExecuteCommand()
   at NuGet.CommandLine.Command.ExecuteCommandAsync()
   at NuGet.CommandLine.Command.Execute()
   at NuGet.CommandLine.Program.MainCore(String workingDirectory, String[] args)
```

This error was fixed in 3.5.0-rc-1365 but then it's as @TimBarham describe about the folders. If I created a packages folder inside the project directory and in that one put the nupkg for the dependency the package was created correctly.

If im using an earlier version of 3.5.0 (https://www.myget.org/feed/nugetbuild/package/nuget/NuGet.CommandLine/3.5.0-rc-1315) then the pack is working as expected so compare what this version is doing during pack with the current source.

All the above versions silence continue if the nupkg for the dependency not was found.

**Comment by toddm** [comment]
This is fixed in Dev, and going into 3.5.0 beta 2.
